src/main/java/springboot/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,99 @@
+package springboot.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import springboot.dto.cartitem.AddCartItemRequestDto;
+import springboot.dto.cartitem.CartItemDto;
+import springboot.dto.cartitem.UpdateCartItemRequestDto;
+import springboot.dto.shoppingcart.ShoppingCartDto;
+import springboot.service.ShoppingCartService;
+
+@Tag(name = "ShoppingCart", description = "Operations related to shopping_carts")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @Operation(
+            summary = "Add item to cart",
+            description = "Add new item to cart. Accessible to USER and ADMIN roles.",
+            responses = {
+                    @ApiResponse(responseCode = "200",
+                            description = "Item added successfully"),
+                    @ApiResponse(responseCode = "403",
+                            description = "Forbidden")
+            }
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping
+    public CartItemDto addToCart(@RequestBody AddCartItemRequestDto requestDto) {
```

Comment:
add Authentication authentication for each endpoint
```suggestion
    public ShoppingCartDto addToCart(Authentication authentication, @RequestBody @Valid AddCartItemRequestDto requestDto) {
```